### PR TITLE
feat: add user public key valid check

### LINKED ISSUE #220: [Reference] Add user public key valid check
Common question in discord is how to figure out if a public key is valid for a specific user (meaning a system account owned account used to interact with other dapps)

Should be both using:
1. `new PublicKey("<address>")`
2. `key.isOnCurve("<address>")`

Let's add this to keypairs and wallets reference section

### PR DESCRIPTION
**Pull request by dboures**
closes #220

### CODE REVIEW

File: code/keypairs-and-wallets/check-valid-publickey/check-valid-publickey.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { PublicKey } from "@solana/web3.js";
+
+// Note that Keypair.generate() will always give a public key that is valid for users
+const key = new PublicKey("5oNDL3swdJJF1g9DzJiZ4ynHXgszjAEpUkxVYejchzrY"); // Valid public key
+console.log(PublicKey.isOnCurve(key.toBytes())); // Lies on the ed25519 curve and is suitable for users
+
+const offCurveAddress = new PublicKey("4BJXYkfvg37zEmBbsacZjeQDpTNx91KppxFJxRqrz48e"); // Valid public key
+console.log(PublicKey.isOnCurve(offCurveAddress.toBytes())); // Not on the ed25519 curve, therefore not suitable for users
+
+const errorPubkey = new PublicKey("testPubkey"); // Is not a valid public key
```
Comment (jacobcreech) [inline code/keypairs-and-wallets/check-valid-publickey/check-valid-publickey.ts:10]: Could you wrap this in:
```javascript
(async function () {

})();
```

That way it is immediately runnable?

**Review by jacobcreech — CHANGES_REQUESTED** [review]
One small change. Otherwise we're good to go!

File: code/keypairs-and-wallets/check-valid-publickey/check-valid-publickey.ts (hunk 1 above)
Comment (dboures) [reply code/keypairs-and-wallets/check-valid-publickey/check-valid-publickey.ts]: Good point, this has been updated.

**Review by jacobcreech — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**